pkg/apis/apps/validation/validation.go

```diff
@@ -39,10 +39,10 @@ import (
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
 func ValidateStatefulSetName(name string, prefix bool) []string {
-	// TODO: Validate that there's name for the suffix inserted by the pods.
+	// TODO: Validate that there's room for the suffix inserted by the pods.
 	// Currently this is just "-index". In the future we may allow a user
 	// specified list of suffixes and we need  to validate the longest one.
-	return apimachineryvalidation.NameIsDNSSubdomain(name, prefix)
+	return apimachineryvalidation.NameIsDNSLabel(name, prefix)
```

Comment:
looks like there's a call chain of ValidateStatefulSetUpdate → ValidateStatefulSet → ValidateStatefulSetName

if someone has a statefulset with a name like this and a finalizer, does this change mean they can't remove the finalizer and delete the statefulset?

Reply:
https://github.com/kubernetes/kubernetes/pull/114510